src/arrayMethodSort.js

```diff
@@ -4,9 +4,33 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let swapped;
+    let n = this.length;
+
+    do {
+      swapped = false;
+
+      for (let i = 0; i < n - 1; i++) {
+        if (compareFunction) {
+          if (compareFunction(this[i], this[i + 1]) > 0) {
+            [this[i], this[i + 1]] = [this[i + 1], this[i]];
+            swapped = true;
+          }
+        } else {
+          if (String(this[i]) > String(this[i + 1])) {
```

Comment:
When converting elements to strings, it's important to be aware that this will not work correctly for all data types. For example, objects will be converted to the string '[object Object]', which might not be the intended behavior. If supporting a wide range of data types is important, consider implementing a more robust type checking and conversion strategy.